### PR TITLE
expression: `sec_to_time` is not compatible with MySQL when the argument is a scientific notation number

### LINKED ISSUE #15613: expression: `sec_to_time` is not compatible with MySQL when the argument is a scientific notation number
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?

```
select sec_to_time(1e-5)
```
### 2. What did you expect to see?
```
mysql> select sec_to_time(1e-5);
+-------------------+
| sec_to_time(1e-5) |
+-------------------+
| 00:00:00.000010   |
+-------------------+
1 row in set (0.00 sec
```

### 3. What did you see instead?
```
mysql> select sec_to_time(1e-5);
ERROR 1292 (22007): Truncated incorrect time value: '00:00:1e-05'
```

### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)
The latest master branch.

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #15613 

### What is changed and how it works?
Fix the issue that `sec_to_time` is not compatible with MySQL when the argument is a scientific notation number.

Convert the scientific notation number to a normal decimal string before parsing.

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: types/convert.go [hunk 1]
```diff
@@ -216,20 +216,30 @@ func convertScientificNotation(str string) (string, error) {
...
+		if point == len(f) { // no point 1E5
+			return f[:point] + strings.Repeat("0", int(exp)), nil
+		}
+
 		// 123.456 >> 5 = 12345600
 		return f[:point] + f[point+1:] + strings.Repeat("0", point+int(exp)-len(f)+1), nil
 	} else { // move point left
 		exp = -exp
 		if int(exp) < point { // 123.456 << 2 = 1.23456
 			return f[:point-int(exp)] + "." + f[point-int(exp):point] + f[point+1:], nil
```
Comment (SunRunAway) [inline types/convert.go]: Will `123E-1` be handled correctly?

Comment (SunRunAway) [reply types/convert.go]: I think we could firstly extract the integer part and the decimal part of `f`, then do the point movement according to `exp`, which makes code more understandable.

File: expression/builtin_time.go [hunk 2]
```diff
@@ -5979,9 +5979,13 @@ func (b *builtinSecToTimeSig) evalDuration(row chunk.Row) (types.Duration, bool,
 		second = seconds % 60
 	}
 	secondDemical = float64(second) + demical
+	secStr, err := types.ConvertScientificNotation(fmt.Sprintf("%v", secondDemical))
```
Comment (SunRunAway) [inline expression/builtin_time.go]: Could `strconv.FormatFloat` be used here?
https://play.golang.org/p/JDZU_za0JoV

Comment (qw4990) [reply expression/builtin_time.go]: A nice suggestion! I have updated this PR to use this method and reverted `convert.go`.

**Comment by qw4990** [comment]
@SunRunAway @XuHuaiyu @wshwsh12  PTAL, this problem may cause the test `TestVectorizedBuiltinTimeFunc` unstable.

**Pull request merged**